libcudacxx/test/libcudacxx/std/text/charconv/charconv.from.chars/integral/int128.pass.cpp

```diff
@@ -0,0 +1,1370 @@
+  cuda::std::strncpy(buff, str, len);
+  buff[len] = '9' + 1;
+  test_from_chars<T>(buff, len + 1, base, len, success_errc, ref_val);
+
+  // 3. Test a string that has invalid characters at the beginning
+  buff[0] = '9' + 1;
+  cuda::std::strncpy(buff + 1, str, len);
+  test_from_chars<T>(buff, len + 1, base, 0, cuda::std::errc::invalid_argument);
+
+  // 3. Test a string that has white characters at the beginning
+  buff[0] = ' ';
+  cuda::std::strncpy(buff + 1, str, len);
+  test_from_chars<T>(buff, len + 1, base, 0, cuda::std::errc::invalid_argument);
+
+  // 5. Test a string that has '+' at the beginning
+  buff[0] = '+';
+  cuda::std::strncpy(buff + 1, str, len);
+  test_from_chars<T>(buff, len + 1, base, 0, cuda::std::errc::invalid_argument);
+
+  // 6. Test a string that has '-' at the beginning (unsigned only)
+  if constexpr (cuda::std::is_unsigned_v<T>)
+  {
+    buff[0] = '-';
+    cuda::std::strncpy(buff + 1, str, len);
+    test_from_chars<T>(buff, len + 1, base, 0, cuda::std::errc::invalid_argument);
+  }
+}
+
+template <class T>
+__host__ __device__ constexpr void test_overflow()
+{
+  constexpr int base = 10;
+
+  TestItem item{};
+
+  // 1. Test `max + 1` overflow
+  {
+    item.str_signed   = "170141183460469231731687303715884105728";
+    item.str_unsigned = "340282366920938463463374607431768211456";
+    test_from_chars<T>(item, base, true);
+  }
+
+  // 2. Test positive huge value overflow
+  {
+    item.str_signed   = "1239098123333333333333333333333333333333333333333333";
+    item.str_unsigned = "1239098123333333333333333333333333333333333333333333";
+    test_from_chars<T>(item, base, true);
+  }
+
+  // 3. Test `min - 1` overflow
+  if constexpr (cuda::std::is_signed_v<T>)
+  {
+    item.str_signed   = "-170141183460469231731687303715884105729";
+    item.str_unsigned = "";
+    test_from_chars<T>(item, base, true);
+  }
+
+  // 4. Test negative huge value overflow
+  if constexpr (cuda::std::is_signed_v<T>)
+  {
+    item.str_signed   = "-1239098123333333333333333333333333333333333333333333";
+    item.str_unsigned = "";
+    test_from_chars<T>(item, base, true);
+  }
+}
+
+template <int Base>
+__host__ __device__ constexpr void test_base()
+{
+  constexpr auto items = get_test_items<Base>();
+
+  for (const auto& item : items)
+  {
+    test_from_chars<__int128_t>(item, Base);
+    test_from_chars<__uint128_t>(item, Base);
+  }
+
+  // Test overflow cases (base 10 only)
+  if constexpr (Base == 10)
+  {
+    test_overflow<__int128_t>();
+    test_overflow<__uint128_t>();
+  }
+}
+
+template <int... Base>
+__host__ __device__ constexpr void test_helper(cuda::std::integer_sequence<int, Base...>)
+{
+  (test_base<Base + first_base>(), ...);
+}
+
+__host__ __device__ constexpr bool test()
+{
+  test_helper(cuda::std::make_integer_sequence<int, last_base - first_base + 1>{});
+
+  return true;
+}
+
+#endif // _CCCL_HAS_INT128()
+
+int main(int, char**)
+{
+#if _CCCL_HAS_INT128()
+  test();
+  static_assert(test());
```

Comment:
Rather than expanding the constexpr limit more and more, why dont we try to move the static_assert one level up and static_assert the individual `test_helper` 

That should also alleviate a lot of pressure on the constepxr evaluator